lib/Sema/SemaBounds.cpp

```diff
@@ -520,22 +518,77 @@ namespace {
             assert("unexpected cast failure");
             return CreateBoundsNone();
           }
-          switch (UO->getOpcode()) {
-            case UnaryOperatorKind::UO_AddrOf: {
-              Expr *SubExpr = UO->getSubExpr();
-              if (SubExpr->getType()->isFunctionType())
-                return CreateBoundsNone();
+          UnaryOperatorKind Op = UO->getOpcode();
 
-              return LValueBounds(SubExpr);
-            }
-            default:
-              // TODO: fill in other cases
+          if (Op == UnaryOperatorKind::UO_AddrOf) {
+            Expr *SubExpr = UO->getSubExpr();
+            if (SubExpr->getType()->isFunctionType())
               return CreateBoundsNone();
+
+              return LValueBounds(SubExpr);
```

Comment:
Your indentation here is odd, it looks like you have a return directly after a return, but i'm pretty sure the logic is sound.